homeassistant/components/logbook/__init__.py

```diff
@@ -328,26 +342,37 @@ def humanify(hass, events, entity_attr_cache):
                     except IndexError:
                         pass
 
-                yield {
+                data = {
                     "when": event.time_fired_isoformat,
                     "name": event_data.get(ATTR_NAME),
                     "message": event_data.get(ATTR_MESSAGE),
                     "domain": domain,
                     "entity_id": entity_id,
                 }
+                if event.context_user_id:
+                    data["context_user_id"] = event.context_user_id
+                if entity_id:
+                    context_entity_id = context_entity_id_map.get(event.context_id)
+                    if context_entity_id and context_entity_id != entity_id:
+                        data["context_entity_id"] = context_entity_id
+                        data["context_entity_id_name"] = _entity_id_name(
```

Comment:
Why not leave it up to the frontend to resolve this?

Reply:
That makes sense with the current PR...  but I'm going to have to refactor to address the other comment so I'll probably end up grabbing it from the event like we currently do since the automation/script/entity may have been deleted.